pkg/repository/chartmuseum.go

```diff
@@ -0,0 +1,375 @@
+		duration = r
+	}
+	return &chartmuseum{
+		instance:  instance,
+		c:         c,
+		ctx:       ctx,
+		logger:    logger,
+		duration:  duration,
+		cancel:    cancel,
+		statusLen: statusLen,
+	}
+
+}
+
+type chartmuseum struct {
+	ctx       context.Context
+	cancel    context.CancelFunc
+	instance  *v1alpha1.Repository
+	duration  time.Duration
+	statusLen int
+
+	c      client.Client
+	logger logr.Logger
+}
+
+func (c *chartmuseum) Start() {
+	c.logger.Info("Start to fetch")
+	go wait.Until(c.Poll, c.duration, c.ctx.Done())
+}
+
+func (c *chartmuseum) Stop() {
+	c.logger.Info("Delete Or Update Repository, stop watcher")
+	c.cancel()
+}
+
+func (c *chartmuseum) Poll() {
+	cond := v1alpha1.Condition{
+		Status:             v1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+	}
+	indexFile, err := c.fetchIndexYaml()
+	if err != nil {
+		c.logger.Error(err, "Failed to fetch index file")
+		cond.Status = v1.ConditionFalse
+		cond.Message = "failed to fetch index file"
+		cond.Reason = v1alpha1.ReasonUnavailable
+		cond.Type = v1alpha1.TypeFailedSync
+	} else {
+		cond.Message = "index yaml synced successfully, createing components"
+		cond.Reason = v1alpha1.ReasonAvailable
+		cond.Type = v1alpha1.TypeSynced
+
+		diffAction, err := c.diff(indexFile)
+		if err != nil {
+			c.logger.Error(err, "failed to get diff")
+			cond.Status = v1.ConditionFalse
+			cond.Message = err.Error()
+			cond.Reason = v1alpha1.ReasonUnavailable
+			cond.Type = v1alpha1.TypeFailedSync
+		} else {
+			for _, item := range diffAction[0] {
+				c.logger.Info("create component", "Component.Name", item.GetName(), "Component.Namespace", item.GetNamespace())
+				if c.Create(&item); err != nil && !errors.IsAlreadyExists(err) {
+					c.logger.Error(err, "failed to create")
+				}
+			}
+			for _, item := range diffAction[1] {
+				c.logger.Info("update component", "Component.Name", item.GetName(), "Component.Namespace", item.GetNamespace())
+				if err := c.Update(&item); err != nil {
+					c.logger.Error(err, "failed to update component status")
+				}
+			}
+		}
+	}
+	i := v1alpha1.Repository{}
+	if err := c.c.Get(c.ctx, types.NamespacedName{Name: c.instance.GetName(), Namespace: c.instance.GetNamespace()}, &i); err != nil {
+		c.logger.Error(err, "try to update repository, but failed to get the latest version.", "Cond", cond)
+	} else {
+		v1alpha1.UpdateCondWithFixedLen(c.statusLen, &i.Status.ConditionedStatus, cond)
+		if err := c.c.Status().Update(c.ctx, &i); err != nil {
+			c.logger.Error(err, "failed to update repository status")
+		}
+	}
+}
+func (c *chartmuseum) Create(component *v1alpha1.Component) error {
+	return c.c.Create(c.ctx, component)
+}
+
+func (c *chartmuseum) Update(component *v1alpha1.Component) error {
+	return c.c.Status().Update(c.ctx, component)
+}
+
+func (c *chartmuseum) Delete(component *v1alpha1.Component) error {
+	return nil
```

Comment:
add a `TODO` or `Won't implement` comment here ?

Reply:
Follow this link to decide https://github.com/kubebb/core/pull/15#discussion_r1208767509 .
If the field is added, then this function needs to be implemented.

Reply:
👌